components/places/android/src/main/java/mozilla/appservices/places/PlacesConnection.kt

```diff
@@ -346,24 +366,30 @@ open class PlacesReaderConnection internal constructor(connHandle: Long) :
     }
 
     override fun getBookmark(guid: String): BookmarkTreeNode? {
-        val rustBuf = rustCall { err ->
-            LibPlacesFFI.INSTANCE.bookmarks_get_by_guid(this.handle.get(), guid, 0.toByte(), err)
-        }
-        try {
-            return rustBuf.asCodedInputStream()?.let { stream ->
-                unpackProtobuf(MsgTypes.BookmarkNode.parseFrom(stream))
+        readQueryCounters.measure {
+            val rustBuf = rustCall { err ->
+                PlacesManagerMetrics.readQueryTime.measure {
+                    LibPlacesFFI.INSTANCE.bookmarks_get_by_guid(this.handle.get(), guid, 0.toByte(), err)
+                }
+            }
+            try {
+                return rustBuf.asCodedInputStream()?.let { stream ->
+                    unpackProtobuf(MsgTypes.BookmarkNode.parseFrom(stream))
+                }
+            } finally {
+                LibPlacesFFI.INSTANCE.places_destroy_bytebuffer(rustBuf)
             }
-        } finally {
-            LibPlacesFFI.INSTANCE.places_destroy_bytebuffer(rustBuf)
         }
     }
 
     override fun getBookmarksTree(rootGUID: String, recursive: Boolean): BookmarkTreeNode? {
         val rustBuf = rustCall { err ->
-            if (recursive) {
-                LibPlacesFFI.INSTANCE.bookmarks_get_tree(this.handle.get(), rootGUID, err)
-            } else {
-                LibPlacesFFI.INSTANCE.bookmarks_get_by_guid(this.handle.get(), rootGUID, 1.toByte(), err)
+            PlacesManagerMetrics.scanQueryTime.measure {
+                if (recursive) {
+                    LibPlacesFFI.INSTANCE.bookmarks_get_tree(this.handle.get(), rootGUID, err)
+                } else {
+                    LibPlacesFFI.INSTANCE.bookmarks_get_by_guid(this.handle.get(), rootGUID, 1.toByte(), err)
+                }
```

Comment:
Should this query be wrapped in some counters, either `read_query_count` or a new `scan_query_count`?